src/journal/journald-manager.c

```diff
@@ -2408,12 +2416,220 @@ static void manager_load_credentials(Manager *m) {
+}
+
+#define MERGE_BOOL_JOURNALD(name, default_value)        \
+        m->config.name =                                \
+                m->config_by_cmdline.name ||            \
+                m->config_by_conf.name ||               \
+                m->config_by_cred.name ||               \
+                default_value;
+
+#define MERGE_NON_NEGATIVE_JOURNALD(name, default_value)                        \
+        m->config.name =                                                        \
+                m->config_by_cmdline.name >= 0 ? m->config_by_cmdline.name :    \
+                m->config_by_conf.name >= 0 ? m->config_by_conf.name :          \
+                m->config_by_cred.name >= 0 ? m->config_by_cred.name :          \
+                default_value;
+
+static void manager_merge_configs(Manager *m) {
+        assert(m);
+
+        /*
+         * From highest to lowest priority: cmdline, conf, cred
+         */
+        manager_merge_storage(m);
+        manager_merge_forward_to_socket(m);
+
+        MERGE_BOOL_JOURNALD(forward_to_syslog, false);
+        MERGE_BOOL_JOURNALD(forward_to_kmsg, false);
+        MERGE_BOOL_JOURNALD(forward_to_console, false);
+        MERGE_BOOL_JOURNALD(forward_to_wall, true);
+
+        MERGE_NON_NEGATIVE_JOURNALD(max_level_store, LOG_DEBUG);
+        MERGE_NON_NEGATIVE_JOURNALD(max_level_syslog, LOG_DEBUG);
+        MERGE_NON_NEGATIVE_JOURNALD(max_level_kmsg, LOG_NOTICE);
+        MERGE_NON_NEGATIVE_JOURNALD(max_level_console, LOG_INFO);
+        MERGE_NON_NEGATIVE_JOURNALD(max_level_wall, LOG_EMERG);
+        MERGE_NON_NEGATIVE_JOURNALD(max_level_socket, LOG_DEBUG);
+}
+
+static void manager_load_config(Manager *m) {
+        assert(m);
+
+        int r;
+
+        manager_set_defaults(m);
+        manager_reset_configs(m);
+
+        manager_load_credentials(m);
+        manager_parse_config_file(m);
+
+        if (!m->namespace) {
+                /* Parse kernel command line, but only if we are not a namespace instance */
+                r = proc_cmdline_parse(parse_proc_cmdline_item, m, PROC_CMDLINE_STRIP_RD_PREFIX);
+                if (r < 0)
+                        log_warning_errno(r, "Failed to parse kernel command line, ignoring: %m");
         }
+
+        manager_merge_configs(m);
+
+        manager_adjust_configs(m);
+}
+
+static void manager_reload_config(Manager *m) {
+        assert(m);
+
+        manager_set_defaults(m);
+
+        m->config_by_conf = JOURNAL_CONFIG_INIT;
+        manager_parse_config_file(m);
+
+        manager_merge_configs(m);
+        manager_adjust_configs(m);
+}
+
+static int manager_reload_journals(Manager *m) {
+        assert(m);
+
+        int r;
+
+        if (m->system_journal && IN_SET(m->config.storage, STORAGE_PERSISTENT, STORAGE_AUTO)) {
+                /* Current journal can continue being used. Update config values as needed. */
+                journal_file_reload(
```

Comment:
hmm, you are eating up the error here. why not propagate it?

it migh tbe ok to eat up errors in some cases, but this should be made explicit by casting the result to `(void)`